pkg/store/pipelineconfig/store.go

```diff
@@ -0,0 +1,238 @@
+	configurationID p4rtapi.PipelineConfigID
+	replay          bool
+}
+
+// WatchOption is a pipelineconfig option for Watch calls
+type WatchOption interface {
+	apply(*watchOptions)
+}
+
+// watchReplyOption is an option to replay events on watch
+type watchReplayOption struct {
+}
+
+func (o watchReplayOption) apply(options *watchOptions) {
+	options.replay = true
+}
+
+// WithReplay returns a WatchOption that replays past changes
+func WithReplay() WatchOption {
+	return watchReplayOption{}
+}
+
+type watchIDOption struct {
+	id p4rtapi.PipelineConfigID
+}
+
+func (o watchIDOption) apply(options *watchOptions) {
+	options.configurationID = o.id
+}
+
+// WithPipelineConfigID returns a Watch option that watches for configurations based on a given pipeline pipelineconfig ID
+func WithPipelineConfigID(id p4rtapi.PipelineConfigID) WatchOption {
+	return watchIDOption{id: id}
+}
+
+type configurationStore struct {
+	pipelineConfigs atomicmap.Map[p4rtapi.PipelineConfigID, *p4rtapi.PipelineConfig]
+}
+
+// NewAtomixStore creates a new Atomix store for device pipeline configurations
+func NewAtomixStore(client primitive.Client) (Store, error) {
+	pipelineConfigsAtomicMap, err := atomixclient.AtomicMap[p4rtapi.PipelineConfigID, *p4rtapi.PipelineConfig](client)("device-provisioner-pipeline-configurations").
+		Codec(generic.GoGoProto[*p4rtapi.PipelineConfig](&p4rtapi.PipelineConfig{})).
+		Get(context.Background())
+
+	if err != nil {
+		return nil, errors.FromAtomix(err)
+	}
+
+	store := &configurationStore{
+		pipelineConfigs: pipelineConfigsAtomicMap,
+	}
+
+	return store, nil
+
+}
+
+func (s *configurationStore) Get(ctx context.Context, id p4rtapi.PipelineConfigID) (*p4rtapi.PipelineConfig, error) {
+	// If the pipeline config is not already in the cache, get it from the underlying primitive.
+	entry, err := s.pipelineConfigs.Get(ctx, id)
+	if err != nil {
+		return nil, errors.FromAtomix(err)
+	}
+
+	// Decode and return the Configuration.
+	configuration := entry.Value
+	if err := decodePipelineConfiguration(entry, configuration); err != nil {
+		return nil, errors.NewInvalid("pipeline config decoding failed: %v", err)
+	}
+	return entry.Value, nil
+}
+
+func (s *configurationStore) Create(ctx context.Context, pipelineConfig *p4rtapi.PipelineConfig) error {
+	if pipelineConfig.ID == "" {
+		return errors.NewInvalid("no pipeline pipelineconfig ID specified")
+	}
+	if pipelineConfig.TargetID == "" {
+		return errors.NewInvalid("no target ID specified")
+	}
+	if pipelineConfig.Revision != 0 {
+		return errors.NewInvalid("cannot create pipeline pipeline config with revision")
+	}
+	if pipelineConfig.Version != 0 {
+		return errors.NewInvalid("cannot create pipeline pipeline config with version")
+	}
+	pipelineConfig.Revision = 1
+	pipelineConfig.Created = time.Now()
+	pipelineConfig.Updated = time.Now()
+	// Create the entry in the underlying map primitive.
+	entry, err := s.pipelineConfigs.Put(ctx, pipelineConfig.ID, pipelineConfig)
```

Comment:
You should use the `Insert` method here. `Put` is idempotent and therefore could either create _or_ update the entry. The `Create` call should not be idempotent. It should either create a new entry or fail.